src/main/mule/sources/external/systems/apis/api-call-confluence.xml

```diff
@@ -32,7 +32,7 @@ output application/json
 vars.outputPayload default [] ++ payload.results]]></ee:set-variable>
 					</ee:variables>
 				</ee:transform>
-				<set-variable value="#[output application/java --- if(payload.size == payload.limit) vars.counter else 0]" doc:name="Update Counter" doc:id="873c8284-6c10-4883-b898-00f920b3f235" variableName="counter"/>
+				<set-variable value="#[output application/java --- if(payload.size == payload.limit) 0 else vars.counter]" doc:name="Update Counter" doc:id="873c8284-6c10-4883-b898-00f920b3f235" variableName="counter"/>

```

Comment:
This pagination logic deserves a deeper review. This change is fine but the pagination is half baked. For instance, https://developer.atlassian.com/server/confluence/pagination-in-the-rest-api/   shows how we can modify the paging setting a limit and even a start (offset). In the current configuration, this is embedded in the URL path. It should be separated as follows:

```
# SDLC Metrics
sdlc:
 ...
  confluence:
    enabled: "false" #Set this to true if confluence metrics is required
    host: "test.atlassian.net"
    port: "443"
    path: "/wiki/rest/api/content/search?next=true&cql=%28type=page%20and%20space=%27MUL%27%29&expand=version,history,history.lastUpdated"
    pagination:
       limit: "1000"
       start: "0"
```
And  then dynamically append these values during the GET on `api-call-confluence.xml` (recursive call subflow)

Reply:
Correct, when testing, the pagination was causing an infinite loop (max recursive calls) - this change was just to break that loop. all the SDLC part requires improvement